### ISSUE TITLE
Snapshot decorator as alternative Selector/Select

### ISSUE THREAD

**Issue by splincode** [ngxs/store#903]
## I'm submitting a...

<pre><code>
[ ] Regression (a behavior that used to work and stopped working in a new release)
[ ] Bug report  
[ ] Performance issue
[x] Feature request
[ ] Documentation issue or request
[ ] Support request => https://github.com/ngxs/store/blob/master/CONTRIBUTING.md
[ ] Other... Please describe:
</code></pre>

## Current behavior

```ts
@State({ 
 name: 'user',
 default: {
  username: 'value'
 }
})
export class UserState {
}
```

```ts
class MyComponent {
  constructor(private store: Store) {
    console.log(store.selectSnapshot(UserState).username);
  }
}
```

## Expected behavior

```ts
@State<UserInterface>({ 
 name: 'user',
 default: {
  username: 'value'
 }
})
export class UserState {
  @Snapshot() public snapshot: UserInterface;
}
```

```ts
class MyComponent {
  constructor(private state: UserState) {
    console.log(state.snapshot.username);
  }
}
```

Or

```ts
class MyComponent {
  @Snapshot(UserState) public user: UserInterface; // getter
}
```

**Comment by ahnpnl** [comment]
I was looking for this one today since there is a @Select but why no decorator for getting snapshot 😀

**Comment by splincode** [comment]
@ahnpnl https://github.com/ngxs-labs/select-snapshot

**Comment by ahnpnl** [comment]
I will wait for 3.5.0 👍

**Comment by splincode** [comment]
You can try install `select-snapshot` now

```
$ npm install @ngxs-labs/select-snapshot
```

**Comment by arturovt** [comment]
@ahnpnl 

I won't be released in the `3.5.0`. Currently it's a labs project so we let users experiment. If they (you) really like this feature - we're happy to move it to the core :)

**Comment by ahnpnl** [comment]
Definitely I would love to have this in core. Actually I want to wait it to be moved into core because our company's policy doesn't want to have experimental packages in production codes. So I would like to wait for it to be in core package to use.

**Comment by arturovt** [comment]
@ahnpnl 

Sure, I understand that. Maybe you could play with this package at your home pet project, find some issue or say that everything alright. I would appreciate your feedback :)

**Comment by ahnpnl** [comment]
Definitely will do, I usually test labs' features with my pet projects at home.

**Comment by splincode** [comment]
Released in 3.5.0
